src/Stratis.Bitcoin/EventBus/IEventBus.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Stratis.Bitcoin.EventBus
+{
+    /// <summary>
+    /// Event Bus interface
+    /// </summary>
+    public interface IEventBus
+    {
+        /// <summary>
+        /// Subscribes to the specified event type with the specified action
+        /// </summary>
+        /// <typeparam name="TEventBase">The type of event</typeparam>
+        /// <param name="action">The Action to invoke when an event of this type is published</param>
+        /// <returns>A <see cref="SubscriptionToken"/> to be used when calling <see cref="Unsubscribe"/></returns>
+        SubscriptionToken Subscribe<TEventBase>(Action<TEventBase> action) where TEventBase : EventBase;
+
+        /// <summary>
+        /// Unsubscribe from the Event type related to the specified <see cref="SubscriptionToken"/>
+        /// </summary>
+        /// <param name="token">The <see cref="SubscriptionToken"/> received from calling the Subscribe method</param>
+        void Unsubscribe(SubscriptionToken token);
+
+        /// <summary>
+        /// Publishes the specified event to any subscribers for the <see cref="TEventBase"/> event type
+        /// </summary>
+        /// <typeparam name="TEventBase">The type of event</typeparam>
+        /// <param name="eventItem">Event to publish</param>
+        void Publish<TEventBase>(TEventBase eventItem) where TEventBase : EventBase;
```

Comment:
I prefer the word Signal (instead of publish) WDYT

Reply:
the fact is that Publish/Subscribe is a standard used for this pattern.
I don't mind renaming this but who's used to work with Publisher/Subscriber pattern is used to this wording